### DISCUSSION TITLE
Alias for Form-Parameter not in openapi.json (v0.100)

### DISCUSSION THREAD

**Discussion by hochstibe** [fastapi/fastapi#9869] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import Annotated

from fastapi import FastAPI, Form, Query


app = FastAPI()


@app.post("/")
def alias(
    schema_: Annotated[
        str | None,
        Form(
            alias="schema",
            description="alias not used in openapi.json",
            example="foo",
        ),
    ] = None,
):
    return {"schema": schema_}


@app.get("/")
def alias(
    schema_: Annotated[
        str | None,
        Query(
            alias="schema",
            description="working as expected",
            example="foo",
        ),
    ] = None,
):
    return {"schema": schema_}
```

### Description

It might be related to #9860 and Pydantic / Openapi. 

Because of Pydantic-v2, a parameter cant be named `schema` anymore and it suggests an alias.

For the Form-Parameter, in the `openapi.json`, the parameter is named `schema_` despite defining the alias. (therefore the documentation is wrong.

Posting a request with curl is working as expected. 
The Query-Parameter is documented as expected in the `openapi.json`

With Fastapi 0.98, the example works as expected.

### Operating System

Windows

### FastAPI Version

0.100.0

### Python Version

3.10.2

**Comment by hochstibe** [comment]
It is related to pydantic v2. With fastapi v0.100 + pydantic v1, it is working as expected

**Comment by Kludex** [comment]
Is it only with Form?

**Reply by hochstibe** [reply]
The Query works, i didn't test others (yet)

**Reply by hochstibe** [reply]
* Form: Alias is not used in openapi.json, the parameter is not used in the function
  * Schema:  `{"Body_alias_form_post":{"properties":{"schema_":{"anyOf":[{"type":"string"},{"type":"null"}],"title":"Schema ","description":"alias not used in openapi.json, parameter not sent to function","examples":["foo"]}},"type":"object","title":"Body_alias_form_post"}`
* Query: Working as expected
  * `{"name":"schema","in":"query","required":false,"schema":{"anyOf":[{"type":"string"},{"type":"null"}],"description":"working as expected","examples":["foo"],"title":"Schema"},"description":"working as expected"}`
* Path: Named as alias in openapi.json, the parameter is not used in the function
  * `{"name":"schema","in":"path","required":false,"schema":{"anyOf":[{"type":"string"},{"type":"null"}],"description":"alias used in schema, but parameter not passed to function","examples":["foo"],"title":"Schema"},"description":"alias used in schema, but parameter not passed to function"}` 
* Header: working as expected
  * `{"name":"schema","in":"header","required":false,"schema":{"anyOf":[{"type":"string"},{"type":"null"}],"description":"working as expected","examples":["foo"],"title":"Schema"},"description":"working as expected"}]`
* Cookie: named as alias in opnapi.json, the parameter is not used in the function
  * `{"name":"schema","in":"cookie","required":false,"schema":{"anyOf":[{"type":"string"},{"type":"null"}],"description":"alias used in schema, but not passed to function","examples":["foo"],"title":"Schema"},"description":"alias used in schema, but not passed to function"}`

**Comment by harol97** [comment]
I have same issue. Have you been able to fix it ?

**Reply by hochstibe** [reply]
I am still on pydantic v1 (because of some issues openapi issues)

**Comment by harol97** [comment]
check [#10319 ](https://github.com/tiangolo/fastapi/pull/10319)

**Comment by Kludex** [comment, marked as answer]
I've created https://github.com/tiangolo/fastapi/issues/10286 to track this issue.
